### PR TITLE
fix: Missing command/option errors are reported better

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/web-ext/issues/169

I also noticed a test was indented wrong -- we didn't have an eslint indent rule. This adds an indent rule (sorry for the noise).

### CODE REVIEW

**Comment by kumar303** [comment]
ooh, since this patch added `yargs.exitProcess(false)` I was able to write a test to cover yargs' magic `--version` handling. This should bring line coverage up to 100% ✨

File: tests/test.program.js [hunk 1]
```diff
@@ -202,17 +249,38 @@ describe('program.main', () => {
...
 
+  it('can get the program version', () => {
+    const fakeVersionGetter = sinon.spy(() => '<version>');
+    const fakeCommands = fake(commands, {
+      build: () => Promise.resolve(),
+    });
+    const projectRoot = '/pretend/project/root';
+    // For some reason, executing --version like this
+    // requires a command. In the real CLI, it does not.
+    return run(['--version', 'build'],
```
Comment (rpl) [inline tests/test.program.js]: @kumar303 I looked a bit deeper into this one and I think that the different behavior is related to the `exitProcess` boolean:

when `web-ext` is executed from the command line `exitProcess` is true and `yargs` makes `web-ext` exit just before the commands are checked:

```
...
    return new Promise(
      (resolve) => {
        this.yargs.exitProcess(this._exitProcess);  // <<-- on `web-ext --verbose` yargs exit here
        argv = this.yargs.argv;
        cmd = argv._[0];
        if (cmd === undefined) {
          throw new WebExtError('No sub-command was specified in the args');
        }
...
```

**Comment by rpl** [comment]
@kumar303 This PR looks good to me
 
I'm thinking that maybe `exitProcess` could be renamed into something that doesn't sounds/looks like `systemProcess` (which has a totally different meanings / roles / values), but I consider it a nit (or a matter of taste) and not a critical issue.

File: tests/test.program.js (hunk 1 above)
Comment (kumar303) [inline tests/test.program.js]: Hmm, I actually cannot reproduce this with an isolated yargs program. It's probably not worth spending too much time on it. Here's what I tried to mimic what web-ext does:
````javascript
var yargs = require('yargs');
yargs
  .help('help')
  .demand(1)
  .strict()
  .exitProcess(false)
  .version(function() {
    return "the version!"
  })
  .command('thing', '', function(yargs) {
  }, function() {
    console.log('running thing');
  })
  .argv;
````

**Pull request merged**